dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/DataOkHttpUploaderV2.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.net
+
+import android.os.Build
+import com.datadog.android.BuildConfig
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.log.Logger
+import java.util.Locale
+import java.util.UUID
+import okhttp3.Call
+import okhttp3.Request
+import okhttp3.RequestBody
+
+internal abstract class DataOkHttpUploaderV2(
+    internal var intakeUrl: String,
+    internal var clientToken: String,
+    internal var source: String,
+    internal val callFactory: Call.Factory,
+    internal val contentType: String,
+    internal val internalLogger: Logger
+) : DataUploader {
+
+    private val uploaderName = javaClass.simpleName
+
+    private val userAgent by lazy {
+        System.getProperty(DataOkHttpUploader.SYSTEM_UA).let {
+            if (it.isNullOrBlank()) {
+                "Datadog/${BuildConfig.SDK_VERSION_NAME} " +
+                    "(Linux; U; Android ${Build.VERSION.RELEASE}; " +
+                    "${Build.MODEL} Build/${Build.ID})"
+            } else {
+                it
+            }
+        }
+    }
+
+    // region DataUploader
+
+    @Suppress("TooGenericExceptionCaught")
+    override fun upload(data: ByteArray): UploadStatus {
+        val requestId = UUID.randomUUID().toString()
+        val uploadStatus = try {
+            executeUploadRequest(data, requestId)
+        } catch (e: Throwable) {
+            internalLogger.e("Unable to upload batch data.", e)
+            UploadStatus.NETWORK_ERROR
+        }
+
+        uploadStatus.logStatus(uploaderName, data.size, devLogger, false, requestId)
+        uploadStatus.logStatus(uploaderName, data.size, internalLogger, true, requestId)
```

Comment:
Is one of these loggers sending data to org2 as part of the Internal Monitoring feature? If so, I think we should ignore some more cases in `logStatus()` as for example there's not much sense to send us error on each `INVALID_TOKEN_ERROR`.